### ISSUE TITLE
Question about heterozygosity

### ISSUE THREAD

**Issue by MartinLaforest** [schatzlab/genomescope#26]
Hello, 

Sorry for the question that may seem obvious to everyone else. The software reports 2.5% heterozygosity for my sample. Does this mean I have 1 base every 40 that is heterozygous?

It seems high, we are at back-cross generation 7.

Can it be overestimated because genome size is underestimated (probably because of repeated sequences).

Thanks for you help.

**Comment by malonge** [comment]
Such heterozygosity should be evident in the coverage vs. frequency plots. Can you share those?

**Comment by mschatz** [comment]
Yes, it would be helpful if you could share the plots. And yes, 2.5% should
mean on average a heterozygous variant every 40bp.

Cheers

Mike

**Comment by MartinLaforest** [comment]
Alright, here are my files. 
I'm very interested by your feedback! 

Thank you!!!

[model.txt](https://github.com/schatzlab/genomescope/files/3729453/model.txt)
![plot log](https://user-images.githubusercontent.com/55760509/66835160-07816c80-ef2d-11e9-94c1-698bad1df87b.png)
![plot](https://user-images.githubusercontent.com/55760509/66835163-09e3c680-ef2d-11e9-9fb3-7e28ecdbd5f8.png)
[progress.txt](https://github.com/schatzlab/genomescope/files/3729455/progress.txt)
[summary.txt](https://github.com/schatzlab/genomescope/files/3729456/summary.txt)

**Comment by mschatz** [comment]
At first glance, the fit looks pretty good and does suggest a high rate of
heterozygosity. Note the tall peak centered at around 19 (heterozygous
kmers) and the next shoulder is centered at around 38 (homozygous kmers).
This method does sometimes get confused if there is a very low rate of
heterozygosity (essentially 0) , in which case the homozygous peak would be
at 19, and there would be a slight heterozygous peak at about 19/2=9.5, but
this is getting lost in the error kmers. Do you have any idea on the
expected genome size? Note the reported size is the haploid genome size (it
reports 3GB for human even though the total diploid genome size is 6GB)

Thanks!

Mike

**Comment by MartinLaforest** [comment]
Thanks Mike,

I went back to the original publication where the genome size was estimated by flow cytometry, the value is 1.135 gb. I notice however that the polidy level is 4... I've read somewhere that your software is better for diploid genomes. That might explain the discrepancy... 

I'm sorry for that. I though that I was working with a diploid plant... It's a weed and sometimes, weeds are peculiar in the sense that ploidy levels varies... :-(

I'm not sure how this changes our discussion! The higher heterozygous rate might just be because it is a tetraploid!

Thanks

**Comment by MartinLaforest** [comment]
Well actually, we don't know if it's a tetraploid... We need to put it under the microscope. We will also do flow cytometry on our sample.

**Comment by mschatz** [comment]
Ah, it turns out we have recently developed a new algorithm that supports
higher ploidy. Can you give this a try:
http://genomescope.org/genomescope2.0

Good luck!

Mike

**Comment by MartinLaforest** [comment]
FAN-TAS-TIC! :-)

I'm now left to understand those results :-)

![Genomescope2-2](https://user-images.githubusercontent.com/55760509/66924502-bdfe5380-eff8-11e9-8f62-4b6bd8969758.PNG)
![Genomescope2-3](https://user-images.githubusercontent.com/55760509/66924503-bdfe5380-eff8-11e9-80ab-1a18a0766f54.PNG)
![Genomescope2-1](https://user-images.githubusercontent.com/55760509/66924504-bdfe5380-eff8-11e9-8fb7-f3b36b966e76.PNG)

**Comment by mschatz** [comment]
The dominate mode of variation is aabb, which is a signature of a
allotetraploid. I think this would also explain why there is still
substantial variation even after all of the rounds of inbreeding. We have
some discussion about this in the "5.3 Allotetraploid vs. Autotetraploid"
section of the preprint for GenomeScope 2.0:
https://www.biorxiv.org/content/10.1101/747568v1

Good luck

Mike
